plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -124,7 +124,7 @@ export class ElasticSearchSearchEngine implements SearchEngine {
     logger.info('Initializing ElasticSearch search engine.');
     return new Client({
       node: config.getString('node'),
-      auth: config.get<ElasticConfigAuth>('auth'),
+      auth: config.getOptional<ElasticConfigAuth>('auth'),
```

Comment:
While at it I'd ❤️ if this was switched to using `getOptionalString` etc. for each field instead. Using the more precise method provides much better error messaging, especially with multiple config files involved.

Reply:
hm, I don't think it makes a difference in this case since every field in the `'auth'` param is also optional

Reply:
It does, since there's validation on the type and structure of the config data as well. Of someone for example adds an array here instead it's gonna blow up pretty late at runtime. Schema validation will catch it but only if it's run

Reply:
@Rugvip you were right! It actually caught a type error while I was working on this change. Turns out that `bearer` auth config is currently not supported in the version of `@elastic/elasticsearch` we have installed:
- Latest released version : [7.13.0](https://www.npmjs.com/package/@elastic/elasticsearch)
- v7.13.0 [type def](https://github.com/elastic/elasticsearch-js/blob/v7.13.0/index.d.ts#L109)

Looks like `bearer` auth support is added in [v7.14](https://github.com/elastic/elasticsearch-js/blob/7.14/index.d.ts#L110) but it hasn't been released yet.

FYI CC: @Xantier @iamEAP